### PR TITLE
Read cidr blocks from the cluster config file for docker

### LINKED ISSUE #219: Pod CIDR in the generated spec is not used for cluster provision
The pod CIDR was updated in the generated spec:

```
clusterNetwork:
  cni: cilium
  pods:
    cidrBlocks:
    - 10.10.0.0/16
```

Despite the changed, `kubectl cluster-info dump --kubeconfig dev-cluster-eks-a-cluster.kubeconfig | grep -i cidr` reported:

```
 "io.cilium.network.ipv4-pod-cidr": "192.168.0.0/24",
                "podCIDR": "192.168.0.0/24",
                "podCIDRs": [
                    "io.cilium.network.ipv4-pod-cidr": "192.168.1.0/24",
                "podCIDR": "192.168.1.0/24",
                "podCIDRs": [
                            "--allocate-node-cidrs=true",
                            "--cluster-cidr=192.168.0.0/16",
```

Is modify the spec the correct way to configure the cluster CIDR and direct Cilium to follow the cluster CIDR?

### PR DESCRIPTION
**Pull request by bnrjee**
*Issue #, if available:*

*Description of changes:*
At the moment, the pod and service cidr values are all hard coded for docker. With this change, these values will be read from the cluster spec file. This should fix #219.
Test output result:

> eks-anywhere bnrjee$ kubectl cluster-info dump --kubeconfig abc/abc-eks-a-cluster.kubeconfig | grep -i cidr
                    "io.cilium.network.ipv4-pod-cidr": "10.10.1.0/24",
                "podCIDR": "10.10.1.0/24",
                "podCIDRs": [
                    "io.cilium.network.ipv4-pod-cidr": "10.10.0.0/24",
                "podCIDR": "10.10.0.0/24",
                "podCIDRs": [
                            "--allocate-node-cidrs=true",
                            "--cluster-cidr=10.10.0.0/16",
I0921 17:04:45.014930       1 range_allocator.go:116] No Secondary Service CIDR provided. Skipping filtering out secondary service addresses.
I0921 17:04:45.115527       1 range_allocator.go:172] Starting range CIDR allocator
I0921 17:04:45.115532       1 shared_informer.go:240] Waiting for caches to sync for cidrallocator
I0921 17:04:45.115535       1 shared_informer.go:247] Caches are synced for cidrallocator 
I0921 17:04:45.120349       1 range_allocator.go:373] Set node abc-sn6jz PodCIDR to [10.10.0.0/24]
I0921 17:04:51.571683       1 range_allocator.go:373] Set node abc-md-0-85c9ccb857-vt6pw PodCIDR to [10.10.1.0/24]
W0921 17:04:47.239883       1 server_others.go:512] detect-local-mode set to ClusterCIDR, but no IPv6 cluster CIDR defined, , defaulting to no-op detect-local for IPv6
W0921 17:04:59.054069       1 server_others.go:512] detect-local-mode set to ClusterCIDR, but no IPv6 cluster CIDR defined, , defaulting to no-op detect-local for IPv6

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by g-gaston — CHANGES_REQUESTED** [review]
I'm surprised we don't have a test case for that
Can we update the current ones/add a new one?

File: pkg/providers/docker/docker_test.go [hunk 1]
```diff
@@ -97,17 +97,34 @@ func TestProviderGenerateDeploymentFileSuccessUpdateMachineTemplate(t *testing.T
...
 			}),
 			wantFileName: "testdata/valid_deployment_expected.yaml",
 		},
+		{
+			testName: "valid config with cidrs",
+			clusterSpec: test.NewClusterSpec(func(s *cluster.Spec) {
+				s.Name = "test-cluster"
+				s.Spec.KubernetesVersion = "1.19"
+				s.Spec.ClusterNetwork.Pods.CidrBlocks = []string{"10.10.0.0/24"}
+				s.Spec.ClusterNetwork.Services.CidrBlocks = []string{"192.168.0.0/16"}
```
Comment (g-gaston) [inline pkg/providers/docker/docker_test.go:114]: can we add multiple blocks in this test?

**Review by g-gaston — APPROVED** [review]

**Review by jiayiwang7 — APPROVED** [review]

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**